### PR TITLE
JAX Wavelet Transforms

### PR DESCRIPTION
**Pull request by CosmoMatt**
@JessWhitney implemented the synthesis transform in JAX, and it is passing regression tests against s2let. There are some things we should do to accelerate this transform but for the time being we can merge this as is :)

### CODE REVIEW

File: s2wav/transforms/analysis_jax.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    sampling: str = "mw",
+    nside: int = None,
+    reality: bool = False,
+    multiresolution: bool = False,
+    filters: Tuple[jnp.ndarray, jnp.ndarray] = None,
+) -> Tuple[jnp.ndarray, jnp.ndarray]:
+    r"""Wavelet analysis from pixel space to wavelet space for complex signals.
+
+    Args:
+        f (np.ndarray): Signal :math:`f` on the sphere with shape :math:`[n_{\theta}, n_{\phi}]`.
```
Comment (JessWhitney) [inline s2wav/transforms/analysis_jax.py:30]: Just a tiny thing - should change np.ndarray to jnp.ndarry in the docs

File: s2wav/transforms/analysis_jax.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            conjugate symmetry of harmonic coefficients. Defaults to False.
+
+        multiresolution (bool, optional): Whether to store the scales at :math:`j_{\text{max}}`
+            resolution or its own resolution. Defaults to False.
+
+    Returns:
+        f_wav (np.ndarray): Array of wavelet pixel-space coefficients
+            with shape :math:`[n_{J}, 2N-1, n_{\theta}, n_{\phi}]`.
+
+        f_scal (np.ndarray): Array of scaling pixel-space coefficients
```
Comment (JessWhitney) [inline s2wav/transforms/analysis_jax.py:60]: And here :)

**Review by JessWhitney — APPROVED** [review]
All the code changes look good, we should keep the docs up to date where we can, I highlighted a couple areas where that'd be easy to do so.

**Comment by JessWhitney** [comment]
Were the tests failing previously because we had Jmin > Jmax? @CosmoMatt

**Comment by CosmoMatt** [comment]
> Were the tests failing previously because we had Jmin > Jmax? @CosmoMatt

So when I updated J_min to be 1 for the testing cases, I forgot to add a flag that skipped the test if J_min was > J in the other tests, like you say :)

File: s2wav/transforms/analysis_jax.py (hunk 1 above)
Comment (CosmoMatt) [reply s2wav/transforms/analysis_jax.py]: Thanks!

File: s2wav/transforms/analysis_jax.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            to None.
+
+        reality (bool, optional): Whether :math:`f \in \mathbb{R}`, if True exploits
+            conjugate symmetry of harmonic coefficients. Defaults to False.
+
+        multiresolution (bool, optional): Whether to store the scales at :math:`j_{\text{max}}`
+            resolution or its own resolution. Defaults to False.
+
+    Returns:
+        f_wav (np.ndarray): Array of wavelet pixel-space coefficients
```
[the remark this answers is not in the document]
Comment (CosmoMatt) [reply s2wav/transforms/analysis_jax.py]: Thanks!

File: s2wav/transforms/analysis_jax.py (hunk 2 above)
Comment (CosmoMatt) [reply s2wav/transforms/analysis_jax.py]: Thanks!

**Pull request merged**